main.cc

```diff
@@ -88,19 +89,19 @@ class Matrix_t : public repl_new
             rowPtr->push_back(*it);
         }
     }
-
     Index_t * nodelet_addr(Index_t i)
     {
         // dereferencing causes migrations
         return (Index_t *)(rows_ + i);
     }
-    
+  
 private:
     Matrix_t(Index_t nrows) : nrows_(nrows)
     {
         nrows_per_nodelet_ = nrows_ + nrows_ % NODELETS();
```

Comment:
Ugh! I had this wrong in previous code. It should be
```nrows_per_nodelet_ = nrows_ / NODELETS() + nrows_ % NODELETS();```

I have changed it in 4.05_repl_constuctor. I think you're going to need the following code too:
```
static inline Index_t n_map(Index_t i) { return i % NODELETS(); }
static inline Index_t r_map(Index_t i) { return I / NODELETS(); }
```

Make those changes and then update the PR.

Reply:
Amended a new commit "Added nmap and rmap" to this PR.